### PR TITLE
Improve `-Rmodule-loading` to show both the path to the source and to the cached file actually loaded

### PR DESCRIPTION
**Pull request by xymus**
Improve the `-Rmodule-loading` feature:
1. For each loaded module, show both the source path (swiftinterface, modulemap, or non-resilient swiftmodule) and the path to the loaded file (adjacent or cached swiftmodule, or pcm file in a cache).
2. Report the loading of a Swift overlay for a module imported indirectly via clang.

### CODE REVIEW

File: test/ModuleInterface/loading-remarks.swift [hunk 1]
```diff
@@ -1,15 +1,57 @@
...
+
+//--- Client.swift
+
+import SwiftDependency
+import SwiftNonResilientDependency
+import DirectMixedDependency
+
+// CHECK: remark: loaded module 'SwiftShims'; source: '{{.*}}/lib/swift/shims/module.modulemap', loaded: '{{.*}}/cache/{{.*}}/SwiftShims-{{.*}}.pcm'
+// CHECK: remark: loaded module 'Swift'; source: '{{.*}}/Swift.swiftmodule', loaded: '{{.*}}/Swift.swiftmodule/arm64-apple-macos.swiftmodule'
+// CHECK: remark: loaded module 'SwiftDependency'; source: '{{.*}}/SwiftDependency.swiftinterface', loaded: '{{.*}}/cache/SwiftDependency-{{.*}}.swiftmodule'
```
Comment (artemcm) [inline test/ModuleInterface/loading-remarks.swift:54]: 😍

**Review by artemcm — APPROVED** [review]
This will be so immensely useful. Thank you!

**Comment by xymus** [comment]
Removed the reference to `arm64-apple-macos` from the test.

@swift-ci Please smoke test

**Comment by xymus** [comment]
Simplified the test further for Windows paths compatibility.

@swift-ci Please smoke test

**Comment by xymus** [comment]
@swift-ci Please smoke test

**Pull request merged**